server/src/lib/challenge.ts

```diff
@@ -31,74 +28,74 @@ export default class Challenge {
   }
 
   getPoints = async (
-    { client_id, params: { challeng_token } }: Request,
+    { client_id, params: { challenge } }: Request,
     response: Response
   ) => {
-    console.log(
-      `[DEBUG] Challenge.getPoints() - client_id:${client_id}, challenge_token:${challeng_token}`
-    );
-    let data = getMockData();
-
-    response.json(data.challengePoint);
+    response.json(await this.model.getPoints(client_id, challenge));
   };
 
   getWeeklyProgress = async (
-    { client_id, params: { challeng_token } }: Request,
+    { client_id, params: { challenge } }: Request,
     response: Response
   ) => {
-    console.log(
-      `[DEBUG] Challenge.getWeeklyProgress - client_id:${client_id}, challenge_token:${challeng_token}`
-    );
-    let data = getMockData();
-
-    response.json(data.weeklyChallenge);
+    const progress = await this.model.getWeeklyProgress(client_id, challenge);
+    const weeklyProgress = {
+      week: progress.week,
+      user: {
+        speak: progress.clip_count,
+        speak_total: progress.week === 2 ? 100 : 200,
+        listen: progress.vote_count,
+        listen_total: progress.week === 2 ? 50 : 100,
+      },
+      team: { invite: progress.teammate_count, invite_total: 50 },
+    };
+    response.json(weeklyProgress);
   };
 
   getTopMembers = async (
-    { client_id, params: { challeng_token, locale, type } }: Request,
+    { client_id, params: { challenge, locale, type }, query }: Request,
     response: Response
   ) => {
-    console.log(
-      `[DEBUG] Challenge.getTopMembers - client_id:${client_id}, locale: ${locale}, challenge_token:${challeng_token}, type:${type}`
+    response.json(
+      await getLeaderboard({
+        dashboard: 'challenge',
+        type: type === 'vote' ? 'vote' : 'clip',
```

Comment:
nice! I had hesitated make it an interface and move it to `common/challenge.ts` since it has string literals like `vote `and `clip`, but hey, maybe we need to reference it in other places later, and type checking is something of value. I will just move it.
Great comment!